workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+```html
+<body>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/p5.js/0.4.12/p5.js"></script>
+</body>
+```
+
+We'll also need to attach our JavaScript file, `main.js`. Add this below the line that adds p5.js:
+
+```html
+<body>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/p5.js/0.4.12/p5.js"></script>
+  <script src="main.js"></script>
+</body>
+```
+
+Now we'll save (with shortcut: `CTRL+s` / `Command+s`) and open Live Preview (`Preview` > `Live Preview`).
+
+### Priming the JS File
+
+Double-click `main.js` to open, and type the following:
+
+```js
+function setup() {
+}
+
+function draw() {
+}
+```
+
+p5.js works by automatically calling two special functions: [`setup()`](http://p5js.org/reference/#/p5/setup) and [`draw()`](http://p5js.org/reference/#p5/draw) to create the visuals on your webpage. We'll be writing our own code in these functions, so that p5.js can then run our code.
+
+`setup()` is run only once, at the beginning. `draw()` on the other hand, is run repeatedly after `setup()` finishes, and in this way, provides the basis for any animation or interaction you see in your project.
+
+Let's save and refresh Live Preview.
+
+It looks like nothing, because our functions do nothing. Let's add something for `setup()` to set up.
+
+```js
+function setup() {
+  createCanvas(480,600);
+}
+```
+
+[`createCanvas()`](http://p5js.org/reference/#p5/createCanvas) is a function that takes two arguments (two numbers for width and height, respectively) and creates an [HTML canvas element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas) (i.e., the thing that will house your project) of that size.
+
+In this project, we'll be creating a tiling of overlapping circles.
+
+First, let's decide how many circles we want in each row. We can store this number in a variable **at the top, above both functions**:
+
+```js
+var NUM_CIRCLES = 12;
+
+function setup() {
+  createCanvas(480,600);
+}
+
+function draw() {
+}
+```
+
+We'll use this to determine the diameter of the circles we want to draw, by doing some simple math. Let's add the declaration at the top of the file, underneath `NUM_CIRCLES`, and define our variable within `setup()`:
+
+```js
+var NUM_CIRCLES = 12;
+var CIRCLE_DIA;
+
+function setup() {
+  createCanvas(480,600);
+  CIRCLE_DIA = width/NUM_CIRCLES;
+}
+```
+
+p5.js stores the width of the canvas in a constant named `width`. By dividing the width by the number of circles, we can calculate the length of the diameter, which we'll store in `CIRCLE_DIA`.
+
+## Part III: Drawing on the Canvas
+
+### Drawing One Circle
+
+p5.js makes drawing ellipses and circles easy with the function [`ellipse()`](http://p5js.org/reference/#p5/ellipse).
+
+Let's see the `ellipse()` function in action by drawing a circle in the middle of the screen:
+
+```js
+function draw() {
+  ellipse(width/2,height/2,CIRCLE_DIA,CIRCLE_DIA);
+}
+```
```

Comment:
Because it makes sense to draw something in `draw()` but I guess this entire sketch could be put in `setup()`.